src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -416,6 +532,36 @@ public void Put(TStructuralType original)
             }
         }
 
+        private static IODataInstanceAnnotationContainer GetInstanceannotationContainer(PropertyInfo propertyInfo,
+        TStructuralType entity, IODataInstanceAnnotationContainer value, bool create)
+        {
+            if (entity == null)
+            {
+                throw Error.ArgumentNull("entity");
+            }
```

Comment:
this is a private method.  if the contract is that entity should never be null, then make this a debug assert rather than an explicit check in release code.